Sources/Networking/Responses/HealthReportResponse.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  HealthReportResponse.swift
+//
+//  Created by Pol Piella on 4/8/25.
+
+import Foundation
+
+enum HealthCheckStatus: String {
+    case passed
+    case failed
+    case warning
+    case unknown
+}
+
+enum HealthCheckType: String {
+    case apiKey = "api_key"
+    case sdkVersion = "sdk_version"
+    case bundleId = "bundle_id"
+    case products = "products"
+    case offerings = "offerings"
+    case offeringsProducts = "offerings_products"
+}
+
+enum ProductStatus: String {
+    case valid = "ok"
+    case couldNotCheck = "could_not_check"
+    case notFound = "not_found"
+    case actionInProgress = "action_in_progress"
+    case needsAction = "needs_action"
+    case unknown
+}
+
+struct PackageHealthReport {
+    let identifier: String
+    let title: String?
+    let status: ProductStatus
+    let description: String
+    let productIdentifier: String
+    let productTitle: String?
+}
+
+struct OfferingHealthReport {
+    let identifier: String
+    let packages: [PackageHealthReport]
+    let status: HealthCheckStatus
+}
+
+struct OfferingsCheckDetails {
+    let offerings: [OfferingHealthReport]
+}
+
+struct BundleIdCheckDetails {
+    let sdkBundleId: String
+    let appBundleId: String
+}
+
+enum HealthCheckDetails {
+    case offeringsProducts(OfferingsCheckDetails)
+    case bundleId(BundleIdCheckDetails)
+    case products(ProductsCheckDetails)
+}
+
+struct ProductsCheckDetails {
+    let products: [ProductHealthReport]
+}
+
+struct ProductHealthReport {
+    let identifier: String
+    let title: String?
+    let status: ProductStatus
+    let description: String
+}
+
+struct HealthCheck {
+    let name: HealthCheckType
+    let status: HealthCheckStatus
+    let details: HealthCheckDetails?
+
+    enum CodingKeys: String, CodingKey {
+        case name
+        case status
+        case details
+    }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        name = try container.decode(HealthCheckType.self, forKey: .name)
+        status = try container.decode(HealthCheckStatus.self, forKey: .status)
+
+        switch name {
+        case .offeringsProducts:
+            details = (try container.decodeIfPresent(OfferingsCheckDetails.self, forKey: .details))
+                .map({ .offeringsProducts($0) })
+        case .bundleId:
+            details = (try container.decodeIfPresent(BundleIdCheckDetails.self, forKey: .details))
+                .map({ .bundleId($0) })
+
+        case .products:
+            details = (try container.decodeIfPresent(ProductsCheckDetails.self, forKey: .details))
+                .map({ .products($0) })
+
+        default:
+            details = nil
+        }
+    }
```

Comment:
This bit is mostly so that if the value isn't present, the call doesn't entirely fail, right? Maybe in this case, we don't really need it? We are trying to find out what could go wrong after all. It might be the case where a high level catch works